File: test-scenarios/disable-comments/dis-mix-005-all-types.ts
/**
 * DIS-MIX-005: All Comment Types in TypeScript
 * This file contains various disable comments that should all be detected
 */

// eslint-disable-next-line
const eslintDisabled = "eslint disabled";

// @ts-ignore
const tsIgnored: number = "not a number";

// @ts-expect-error
const tsExpectError: string = 123;

/* eslint-disable */
const multiLineEslintDisabled = "multi-line disabled";
/* eslint-enable */

// @ts-nocheck would be at file level, simulating with comment
// Note: @ts-nocheck disables all type checking for the file

export { eslintDisabled, tsIgnored, tsExpectError, multiLineEslintDisabled };

File: test-scenarios/disable-comments/dis_mix_005_all_types.py
"""
DIS-MIX-005: All Comment Types in Python
This file contains various disable comments that should all be detected
"""

import os  # noqa: F401
import sys  # noqa

# fmt: off
badly_formatted_code=1+2+3
# fmt: on

# type: ignore
wrong_type: str = 123

# pylint: disable=invalid-name
BadlyNamedVariable = "should be snake_case"
# pylint: enable=invalid-name

# pragma: no cover
def uncovered_function():
    """This function is excluded from coverage."""
    pass


# mypy: ignore-errors
class IgnoredClass:
    """Class with mypy errors ignored."""

    def __init__(self):
        self.value = "test"
